### ISSUE TITLE
GoFSH should support creating inline instances that have ids that are numbers

### ISSUE THREAD

**Issue by jafeltra** [FHIR/GoFSH#203]
_*Note:* This issue is currently blocked by [this bug](https://github.com/FHIR/sushi/issues/1184) in SUSHI. If SUSHI cannot assign instances with an id that is a number, it is good that GoFSH does not create instances with ids that are numbers. We should not fix this bug before the SUSHI bug is fixed._

Currently, when GoFSH tries to create an inline instance, it checks if the Instance's `id` is a number. If it is a number, it does not set the `Instance` keyword to that numeric id and instead auto-generates a string to be used with the `Instance` keyword. It then creates an assignment rule that sets the id to the numeric value. This allows SUSHI to use the string that is used with the `Instance` keyword when assigning the inline instance, but the actual instance JSON preserves the correct `id`.

For example, when the following JSON is converted to FSH, the `id` is set via an assignment rule and an auto-generated string is used with the `Instance` keyword.

```
{
  "resourceType": "Condition",
  "id": "ExampleCondition",
  "contained": [
    {
      "resourceType": "Patient",
      "id": "123",
      "name": [
        {
          "given": [
            "Jane"
          ],
          "family": "Doe"
        }
      ]
    }
  ],
  "subject": {
    "reference": "#123"
  }
}
```

This behavior is likely implemented because SUSHI cannot currently assign instances that have a numeric id. Even though FHIR allows an `id` to just be a number, SUSHI does not properly recognize the number as an `id` of a resource and instead just thinks it is a number. This is a separate issue [here](https://github.com/FHIR/sushi/issues/1184). Once that issue is fixed, GoFSH does not need to prevent numeric `id`s from being set with the `Instance` keyword.
